woodwork/table_accessor.py

```diff
@@ -145,6 +145,30 @@ def __getattr__(self, attr):
         else:
             raise AttributeError(f"Woodwork has no attribute '{attr}'")
 
+    def __getitem__(self, key):
+        if isinstance(key, list):
+            index = pd.Index(key)
+            columns = self._dataframe.columns
+            in_columns = index.isin(columns).all()
+
+            if not in_columns:
+                diff = list(index.difference(columns).sort_values())
```

Comment:
If you move `diff`'s definition out of the if statement, you can then check `if diff` before raising the error, I think, which could be good if there's a lot of columns so we don't have to check twice for `in_columns` and then `diff`